src/sentry/utils/auth.py

```diff
@@ -247,7 +253,7 @@ def has_completed_sso(request: HttpRequest, organization_id: int) -> bool:
 
 
 def find_users(
-    username: str, with_valid_password: bool = True, is_active: bool | None = None
+    username: str, with_valid_password: bool = True, is_active: Optional[bool] = None
```

Comment:
Should `with_valid_password` be `Optional` given that it has a default value?

Reply:
So `Optional` is really just short hand for `Union[None, ...]`
So defaulting doesn't necessarily mean optional

Reply:
ah so we're saying the _value_ of `with_valid_password` is always a boolean - I was thinking about the value of the passed in argument. makes sense!